Bootstrap failure

https://github.com/Rust-GCC/gccrs/actions/runs/4295444084/jobs/7485866143

```
In file included from ../../gcc/rust/parse/rust-parse.h:734,
                 from ../../gcc/rust/rust-session-manager.cc:23:
../../gcc/rust/parse/rust-parse-impl.h: In instantiation of ‘std::unique_ptr<Rust::AST::ExternalTypeItem> Rust::Parser<ManagedTokenSource>::parse_external_type_item(Rust::AST::Visibility, Rust::AST::AttrVec) [with ManagedTokenSource = Rust::Lexer; Rust::AST::AttrVec = std::vector<Rust::AST::Attribute>]’:
../../gcc/rust/parse/rust-parse-impl.h:6043:14:   required from ‘std::unique_ptr<Rust::AST::ExternalItem> Rust::Parser<ManagedTokenSource>::parse_external_item() [with ManagedTokenSource = Rust::Lexer]’
../../gcc/rust/parse/rust-parse-impl.h:5837:56:   required from ‘std::unique_ptr<Rust::AST::ExternBlock> Rust::Parser<ManagedTokenSource>::parse_extern_block(Rust::AST::Visibility, Rust::AST::AttrVec) [with ManagedTokenSource = Rust::Lexer; Rust::AST::AttrVec = std::vector<Rust::AST::Attribute>]’
../../gcc/rust/parse/rust-parse-impl.h:1238:11:   required from ‘std::unique_ptr<Rust::AST::VisItem> Rust::Parser<ManagedTokenSource>::parse_vis_item(Rust::AST::AttrVec) [with ManagedTokenSource = Rust::Lexer; Rust::AST::AttrVec = std::vector<Rust::AST::Attribute>]’
../../gcc/rust/parse/rust-parse-impl.h:1071:14:   required from ‘std::unique_ptr<Rust::AST::Item> Rust::Parser<ManagedTokenSource>::parse_item(bool) [with ManagedTokenSource = Rust::Lexer]’
../../gcc/rust/parse/rust-parse-impl.h:414:41:   required from ‘std::vector<std::unique_ptr<Rust::AST::Item> > Rust::Parser<ManagedTokenSource>::parse_items() [with ManagedTokenSource = Rust::Lexer]’
../../gcc/rust/parse/rust-parse-impl.h:443:51:   required from ‘std::unique_ptr<Rust::AST::Crate> Rust::Parser<ManagedTokenSource>::parse_crate() [with ManagedTokenSource = Rust::Lexer]’
../../gcc/rust/rust-session-manager.cc:484:62:   required from here
../../gcc/rust/parse/rust-parse-impl.h:4016:5: error: redundant move in initialization [-Werror=redundant-move]
 4016 |     new AST::ExternalTypeItem (std::move (alias_name_tok->get_str ()),
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 4017 |                                std::move (vis), std::move (outer_attrs),
      |                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 4018 |                                std::move (locus)));
      |                                ~~~~~~~~~~~~~~~~~~
```

--- a/gcc/rust/parse/rust-parse-impl.h
+++ b/gcc/rust/parse/rust-parse-impl.h
@@ -4013,9 +4013,8 @@ Parser<ManagedTokenSource>::parse_external_type_item (AST::Visibility vis,
     return nullptr;
 
   return std::unique_ptr<AST::ExternalTypeItem> (
-    new AST::ExternalTypeItem (std::move (alias_name_tok->get_str ()),
-			       std::move (vis), std::move (outer_attrs),
-			       std::move (locus)));
+    new AST::ExternalTypeItem (alias_name_tok->get_str (), std::move (vis),
+			       std::move (outer_attrs), std::move (locus)));
 }
 
 // Parses a "type alias" (typedef) item.

